lib/charms/tempo_k8s/v0/tracing.py

```diff
@@ -484,7 +484,7 @@ def is_ready(self, relation: Optional[Relation] = None):
         """Is this endpoint ready?"""
         relation = relation or self._relation
         if not relation:
-            logger.error(f"no relation on {self._relation_name}: tracing not ready")
+            logger.info(f"no relation on {self._relation_name !r}: tracing not ready")
```

Comment:
Since tracing is an optional relation, I wonder if this should be `debug`. Do we need this log line at all?

Reply:
"log everything" 
cit. @simskij

Reply:
A charm can have many relations but we do not log every charm reinit a "not ready" message for every relation.
Shouldn't a positive log suffice? I.e. "relation added", "relation removed" instead of "no relation".

Reply:
this is not for every relation though: it's only once if there is no relation. 
it may be logged multiple times if the charm checks tracing readiness more than once, we could avoid that. But I think it's useful debug info to know that the charm couldn't find a tracing relation and therefore tracing is disabled.